### PR TITLE
Added latest Rc Smoothing changes

### PR DESCRIPTION
**Pull request by Asizon**
This PR goes with part of https://github.com/betaflight/betaflight/pull/10727 and now is 100% coordination finished with this PR changes.
Added Rc smoothing changes maintaining the compatibility and functions of the previous one. I have used some constants but im not sure were they need to go @McGiverGim . I also use text replaces for i18n, but we can think on create new html group and hide the old, what do you think guys?

image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/tabs/receiver.js [hunk 1]
```diff
@@ -524,23 +533,36 @@ TABS.receiver.initialize = function (callback) {
         $(".bind_btn").toggle(showBindButton);
 
         // RC Smoothing
+        let smoothingOnOff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingMode :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:536]:
```suggestion
        let smoothingOnOff = semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44) ? FC.RX_CONFIG.rcSmoothingMode : FC.RX_CONFIG.rcSmoothinType;
```

File: src/js/tabs/receiver.js [hunk 2]
```diff
@@ -524,23 +533,36 @@ TABS.receiver.initialize = function (callback) {
...
+            FC.RX_CONFIG.rcSmoothingType);
+        let setPointCutoff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingSetpointCutoff :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:538]: Same as above

File: src/js/tabs/receiver.js [hunk 3]
```diff
@@ -524,23 +533,36 @@ TABS.receiver.initialize = function (callback) {
...
+            FC.RX_CONFIG.rcSmoothingInputCutoff);
+        let feedforwardCutoff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingFeedforwardCutoff :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:540]: Same as above

File: src/js/tabs/receiver.js [hunk 4]
```diff
@@ -524,23 +533,36 @@ TABS.receiver.initialize = function (callback) {
...
+            FC.RX_CONFIG.rcSmoothingDerivativeCutoff);
+        let autoFactor = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingAutoFactor :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:542]: Same as above

File: src/js/tabs/receiver.js [hunk 5]
```diff
@@ -524,23 +533,36 @@ TABS.receiver.initialize = function (callback) {
...
             const rcSmoothingNumberElement = $('input[name="rcSmoothingInputHz-number"]');
             const rcSmoothingDerivativeNumberElement = $('input[name="rcSmoothingDerivativeCutoff-number"]');
-            rcSmoothingNumberElement.val(FC.RX_CONFIG.rcSmoothingInputCutoff);
-            rcSmoothingDerivativeNumberElement.val(FC.RX_CONFIG.rcSmoothingDerivativeCutoff);
+            rcSmoothingNumberElement.val(setPointCutoff);
+            rcSmoothingDerivativeNumberElement.val(feedforwardCutoff);
             $('.tab-receiver .rcSmoothing-input-cutoff').show();
             $('select[name="rcSmoothing-input-manual-select"]').val("1");
-            if (FC.RX_CONFIG.rcSmoothingInputCutoff == 0) {
+            if (setPointCutoff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:565]:
```suggestion
            if (setPointCutoff === 0) {
```

File: src/js/tabs/receiver.js [hunk 6]
```diff
@@ -550,14 +572,14 @@ TABS.receiver.initialize = function (callback) {
...
-                    rcSmoothingNumberElement.val(FC.RX_CONFIG.rcSmoothingInputCutoff);
+                    rcSmoothingNumberElement.val(setPointCutoff);
                     $('.tab-receiver .rcSmoothing-input-cutoff').show();
                 }
             }).change();
 
             $('.tab-receiver .rcSmoothing-derivative-cutoff').show();
             $('select[name="rcSmoothing-input-derivative-select"]').val("1");
-            if (FC.RX_CONFIG.rcSmoothingDerivativeCutoff == 0) {
+            if (feedforwardCutoff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:582]:
```suggestion
            if (feedforwardCutoff === 0) {
```

File: src/js/tabs/receiver.js [hunk 7]
```diff
@@ -865,6 +881,15 @@ TABS.receiver.updateRcInterpolationParameters = function () {
 };
 
 function updateInterpolationView() {
+    let smoothingOnOff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingMode :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:884]: Same as above

File: src/js/tabs/receiver.js [hunk 8]
```diff
@@ -873,25 +898,48 @@ function updateInterpolationView() {
     $('.tab-receiver .rcSmoothing-derivative-manual').show();
     $('.tab-receiver .rcSmoothing-input-manual').show();
     if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_42)) {
-        if (FC.RX_CONFIG.rcSmoothingDerivativeCutoff == 0 || FC.RX_CONFIG.rcSmoothingInputCutoff == 0) {
+        if (feedforwardCutoff == 0 || setPointCutoff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:901]:
```suggestion
        if (feedforwardCutoff === 0 || setPointCutoff === 0) {
```

File: src/js/tabs/receiver.js [hunk 9]
```diff
@@ -873,25 +898,48 @@ function updateInterpolationView() {
...
+        $('.rcSmoothingOff').text('OFF');
+        $('.rcSmoothingOn').text('ON');
+        $('.rcSmoothingSetpointTypeSelect').text(i18n.getMessage('receiverRcSmoothingSetpointTypeSelect'));
+        $('.rcSmoothingFeedforwardTypeSelect').text(i18n.getMessage('receiverRcSmoothingFeedforwardTypeSelect'));
+        $('.rcSmoothingSetpointCutoffHz').text(i18n.getMessage('receiverRcSmoothingSetpointCutoffHz'));
+        $('.rcSmoothingFeedforwardCutoffHz').text(i18n.getMessage('receiverRcSmoothingFeedforwardCutoffHz'));
+        $('.rcSmoothingAutoFactorTitle').text(i18n.getMessage('receiverRcSmoothingAutoFactor'));
+    }
+
+    if (smoothingOnOff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:921]:
```suggestion
    if (smoothingOnOff === 0) {
```

File: src/js/tabs/receiver.js [hunk 10]
```diff
@@ -873,25 +898,48 @@ function updateInterpolationView() {
...
+            $('.tab-receiver .rcSmoothing-input-cutoff').hide();
+            $('.tab-receiver .rcSmoothing-derivative-type').hide();
+            $('.tab-receiver .rcSmoothing-input-type').hide();
+            $('.tab-receiver .rcSmoothing-derivative-manual').hide();
+            $('.tab-receiver .rcSmoothing-input-manual').hide();
+            $('.tab-receiver .rcSmoothing-auto-smoothness').hide();
+        }
     }
-    if (FC.RX_CONFIG.rcSmoothingDerivativeCutoff == 0) {
+    if (feedforwardCutoff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:939]:
```suggestion
    if (feedforwardCutoff === 0) {
```

File: src/js/tabs/receiver.js [hunk 11]
```diff
@@ -873,25 +898,48 @@ function updateInterpolationView() {
...
         $('.tab-receiver .rcSmoothing-derivative-cutoff').hide();
     }
-    if (FC.RX_CONFIG.rcSmoothingInputCutoff == 0) {
+    if (setPointCutoff == 0) {
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:942]:
```suggestion
    if (setPointCutoff === 0) {
```

File: src/js/tabs/receiver.js [hunk 12]
```diff
@@ -865,6 +881,15 @@ TABS.receiver.updateRcInterpolationParameters = function () {
 };
 
 function updateInterpolationView() {
+    let smoothingOnOff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingMode :
+        FC.RX_CONFIG.rcSmoothingType);
+    let setPointCutoff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingSetpointCutoff :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:886]: same as above

File: src/js/tabs/receiver.js [hunk 13]
```diff
@@ -865,6 +881,15 @@ TABS.receiver.updateRcInterpolationParameters = function () {
...
+        FC.RX_CONFIG.rcSmoothingInputCutoff);
+    let feedforwardCutoff = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingFeedforwardCutoff :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:888]: same as above

File: src/js/tabs/receiver.js [hunk 14]
```diff
@@ -865,6 +881,15 @@ TABS.receiver.updateRcInterpolationParameters = function () {
...
+        FC.RX_CONFIG.rcSmoothingDerivativeCutoff);
+    let autoFactor = ((semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) ? FC.RX_CONFIG.rcSmoothingAutoFactor :
```
Comment (haslinghuis) [inline src/js/tabs/receiver.js:890]:
```suggestion
    let autoFactor = semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44) ? FC.RX_CONFIG.rcSmoothingAutoFactor : FX.RX_CONFIG.rcSmoothingAutoSmoothness;
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Comment by haslinghuis** [comment]
Please also solve the sonar complaints, that's why the Windows build is failing...

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by haslinghuis — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by McGiverGim** [comment]
I prefer always to have all the elements in the HTML and hide the elements that don't need to be here. It seems to me that needs less maintenance that to have a text totally dynamic.
For the rest, it seems good to me.

File: src/js/tabs/receiver.js [hunk 15]
```diff
@@ -873,25 +896,48 @@ function updateInterpolationView() {
...
         $('.tab-receiver .rcSmoothing-derivative-type').hide();
         $('.tab-receiver .rcSmoothing-input-type').hide();
-        $('.tab-receiver .rcSmoothing-derivative-manual').hide();
-        $('.tab-receiver .rcSmoothing-input-manual').hide();
-        $('.tab-receiver .rcSmoothing-auto-smoothness').hide();
+        $('.tab-receiver .rc-smoothing-channels').hide();
+        $('.tab-receiver input[name="rcSmoothingAutoSmoothness-number"]').attr("max", "250");
+        $('.rcSmoothingTypeMode').text(i18n.getMessage('receiverRcSmoothingMode'));
+        $('.rcSmoothingOff').text('OFF');
+        $('.rcSmoothingOn').text('ON');
```
Comment (McGiverGim) [inline src/js/tabs/receiver.js:911]: Esto debe ir al i18n?

**Comment by Asizon** [comment]
After some talks with @McGiverGim I have reanmed old wrong named features, also removed unneeded MSP changes. Basically 80% of the PR are a renames.

File: src/js/tabs/receiver.js (hunk 15 above)
Comment (haslinghuis) [reply src/js/tabs/receiver.js:892]: ON and OFF should be translated.

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**